kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/ApplicationCommandRegistry.kt

```diff
@@ -423,63 +221,35 @@ public open class ApplicationCommandRegistry : KoinComponent {
         }
 
         val name = command.getTranslatedName(locale)
+        val description = command.getTranslatedDescription(locale)
 
         val response = if (guild == null) {
             // We're registering global commands here, if the guild is null
 
-            kord.createGlobalMessageCommand(name) {
+            kord.createGlobalChatInputCommand(name, description) {
                 logger.trace { "Adding/updating global ${command.type.name} command: $name" }
 
                 this.register(locale, command)
             }
         } else {
             // We're registering guild-specific commands here, if the guild is available
 
-            guild.createMessageCommand(name) {
+            guild.createChatInputCommand(name, description) {
                 logger.trace { "Adding/updating guild-specific ${command.type.name} command: $name" }
 
                 this.register(locale, command)
             }
         }
 
-        try {
-            if (guild != null) {
-                kord.editApplicationCommandPermissions(kord.resources.applicationId, guild.id, response.id) {
-                    command.allowedUsers.map { user(it, true) }
-                    command.disallowedUsers.map { user(it, false) }
-
-                    command.allowedRoles.map { role(it, true) }
-                    command.disallowedRoles.map { role(it, false) }
-                }
-
-                logger.trace { "Applied permissions for command: ${command.name} ($command)" }
-            } else {
-                logger.warn { "Applying permissions to global application commands is currently not supported." }
-            }
-        } catch (e: KtorRequestException) {
-            logger.error(e) {
-                "Failed to apply application command permissions. This command will not be registered." +
-                    if (e.error?.message != null) {
-                        "\n        Discord error message: ${e.error?.message}"
-                    } else {
-                        ""
-                    }
-            }
-        } catch (t: Throwable) {
-            logger.error(t) {
-                "Failed to apply application command permissions. This command will not be registered."
-            }
-
-            return null
-        }
-
-        messageCommands[response.id] = command
+        injectPermissions(guild, command, response.id) ?: return null
 
-        return command
+        return response.id
     }
 
-    /** Register a slash command. **/
-    public open suspend fun register(command: SlashCommand<*, *>): SlashCommand<*, *>? {
+    /**
+     * Creates a KordEx [UserCommand] as discord command and returns the created command's id as [Snowflake].
```

Comment:
Something like this would be more appropriate:

```
Register the given [UserCommand], syncing it to Discord.

@returns Snowflake representing the registered command's ID.
```

This is true for the other similar functions below.

Also, do these really need to have different names? The overloaded approach with the function named `register` was much clearer, in my opinion. If not, at least remove the `Discord` tautology and change it to, eg, `registerUserCommand`.

Reply:
No, it actually does exactly what the name says. It `creates` a `Discord` command of the given type. It does not do *anything* with the registry. This is solely used to create the commands on discord's backend.

Reply:
That doesn't make sense to me, API-wise - if you're going to tell Discord about the command then you're going to want to handle it as well, so it doesn't seem sensible for this to not touch the registry.

Reply:
Previously you had this somewhere in your code and it was duplicated code, I just moved it to a designated function. And it still makes sense in that scope, becaue most subtypes would want to register application commands in Discord's backend at some point.

Reply:
Ah, okay, I see what you mean.